benchmarks/inference/llm-bench/src/llm_bench/clients/azure_ml_client.py

```diff
@@ -0,0 +1,69 @@
+import json
+import requests
+from typing import Any, Dict
+
+from .base import BaseClient
+from ..config import BaseConfigModel
+from ..prompt import Prompt
+from ..response import Response
+from ..status import Status
+
+
+class AzureMLClientConfig(BaseConfigModel):
+    # TODO: Add descriptions for each field
+    api_url: str = ""
+    api_key: str = ""
+    deployment_name: str = ""
+
+
+class AzureMLClient(BaseClient):
+    def __init__(self, config: AzureMLClientConfig) -> None:
+        super().__init__(config)
+        self.api_url = config.api_url
+        self.api_key = config.api_key
+        self.deployment_name = config.deployment_name
+
+    # TODO: Implement these methods
+    def start_service(self) -> Status:
+        pass
+
+    def stop_service(self) -> Status:
+        pass
+
+    def prepare_request(self, prompt: Prompt) -> Dict[str, Any]:
+        if prompt.streaming:
+            raise ValueError("AzureMLClient does not support streaming prompts.")
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": ("Bearer " + self.api_key),
+            "azureml-model-deployment": self.deployment_name,
+        }
+        pload = {
```

Comment:
As a forward looking TODO, do you think we'd ever want to support the chat-completion API?